libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -348,3 +374,100 @@ bool StaticAnalyzer::visit(FunctionCall const& _functionCall)
 	}
 	return true;
 }
+
+void StaticAnalyzer::checkDoubleStorageAssignment(Assignment const& _assignment)
+{
+	size_t storageToStorageCopies = 0;
+	size_t toStorageCopies = 0;
+	size_t storageByteArrayPushes = 0;
+	size_t storageByteAccesses = 0;
+	auto count = [&](TupleExpression const& _lhs, TupleType const& _rhs, auto _recurse) -> void {
+		TupleType const& lhsType = dynamic_cast<TupleType const&>(*type(_lhs));
+		TupleExpression const* lhsResolved = dynamic_cast<TupleExpression const*>(resolveOuterUnaryTuples(&_lhs));
+
+		solAssert(lhsResolved && lhsResolved->components().size() == lhsType.components().size());
```

Comment:
This assert has now been strengthened to make sure that `lhsResolved` is not a nullptr.